README.md

```diff
@@ -1,37 +1,78 @@
 # cppdep
 [![Build Status](https://travis-ci.org/cgilling/cppdep.svg?branch=master)](https://travis-ci.org/cgilling/cppdep)
 
-Playing Around With C++ Dependencies and simple naive compilation
-
-## Warning
-
-This is still in active development. Where it is now in a working state, the config file and command line interface is likely to change in the near future.
+Simple compilation of an arbitrary C++ source tree with autmatic dependency discovery.
 
 ## Goal
-The goal is to create a dependency tree of all the c/c++ source code
-in a directory and its subdirectories. After this, using a few basic
-rules, then automatically being able to build all the binaries defined
-within the source tree.
+Given an arbitrary source tree, `cppdep` is able to scan dependencies and determine which files need to be compiled into binaries. With simple configuration installed libraries can be linked against as well.
+
+## Assumptions
+In order to automatically determine all source files needed to compile a binary, the following assumptions are made about the source tree.
+* All base filenames are unique (i.e there will not be both an a/file.h and b/file.h)
+* If a file includes file.h, then the compiled binary will also need file.cc
+* All includes of files contained within the source tree use double quotes rather that angle brackets
+* All header files have one of the following extensions: `.h`, `.hpp`, `.hh`, `.hxx`
+* All source files have one of the following extensions: `.cc`, `.cxx`, `.c`
 
 ## Usage
 
 ```shell
-cppdep [--config CONFIG_PATH] [--fast] [--concurrency VALUE] [--regex] [--src SRCDIR] [BINARY_NAME]
+cppdep [--config CONFIG_PATH] [--fast] [--concurrency|-c VALUE] [BINARY_NAME]*
 ```
 * `--config`: path to the yaml config file defining the parameters for the build. If not provided $CWD and all parent directories in order will be seaches for a cppdep.yml file.
 * `--fast`: Enable fast include scanning. This means that scanning a file for include statements will stop as soon as a line is found that is not a preprocessor statement, comment, or empty line. (This significantly speeds up the dependency phase – on my test machine/source tree generation time goes from 5s to 0.3s)
 * `--concurrency`: maximum number of concurrent compiles. Also controls the number of files that will be concurrently scanned for dependencies. Because dependency scanning is CPU intensive, it is recommended to test and see what value of `GOMAXPROCS` is appropriate. (Somewhere between 2-4 seems about right based on my testing)
```

Comment:
Does the user need to set GOMAXPROCS themselves or does the --concurrency flag set that? Would be nice to have that get set by the flag itself.